### ISSUE TITLE
Are there any drawbacks to using a simple SSH tunnel instead of V2Ray/Xray/Other advanced techniques?

### ISSUE THREAD

**Issue by burntheplanet** [net4people/bbs#191]
Recently I realized that the best protocol that works almost flawlessly on my private server with most ISPs in Iran is actually VPS. Vless + XTLS + Vision or Trojan + XTLS barely work on Irancell, but SSH works without any packet loss or throttling. However, it seems like everyone recommends either V2Ray or Xray. So, are there any disadvantages to using just a simple SSH tunnel? Can they distinguish it from a regular ssh usage scenario? If not, then considering SSH works the best in terms of speed and reliability, is Xray/V2Ray still the recommended solution (for me)?
Thanks

**Comment by abraham-sah** [comment]
So is my inquiry as well! 

Isn't SSH proxy better for personal use?

Like they can't put a server in black list only because of having SSH traffic.

And It's already well encrypted!

**Comment by Soberia** [comment]
It depends on your ISP. Some ISPs throttle SSH traffic or drop the SSH session after some passed time for at least some of the cloud providers. Therefore, it almost impossible to use SSH in some situations.

Furthermore, you usually don't want to share your SSH credentials with someone else to just help them to overcome the censorship. You may create a non-privilaged user for them, but they still access to so many things.
Proxy tools usually have a built-in user management system to add/remove users on the fly without restarting the service and interrupting the other users.

They also provide more throughput on the same bandwidth in comparison to SSH.

**Comment by burntheplanet** [comment]
Oh, I see. Apparently, for my ISP (Irancell TD-LTE), it seems like it is not throttled whatsoever, since I'm getting around the same speed as native, with around 0% packet loss and connection lasting for at least 3+ hours as tested so far. The only issue might be user management, which is fine for me since I am the only one using it.
On a technical term tho, how are other solutions hypothetically able to provide more bandwidth than SSH? Don't they have stronger and longer encryptions and obfuscations than SSH?

**Comment by Soberia** [comment]
I'm not sure why but I think it must be related to their implementations. In my tests with Irancell LTE, I reached 120Mb with Trojan over xTLS and 80Mb with SSH (both proxied in `v2rayN`)

**Comment by arandomgstring** [comment]
@burntheplanet

To be honest with you, a huge traffic coming from SSH is not very usual. Administrators and developers usually upload (send) noticeable traffic over ssh to their server using SCP or whatever, but they tend to not get huge files from their servers, at least I cannot think of any scenario that is needed (they might use SFTP though, which is ssl based, not ssh based). Therefore, personally, I think when it comes to downloading large files, or watching movies, perhaps it is better to use https rather than ssh. Other than that, I see no other benefit. As for restricting user's permission on shell, you can unset Path variable for them.

**Comment by burntheplanet** [comment]
@arandomgstring yeah, that is true. But I guess in the end, their suspicion just leads to the IP getting blacklisted. And since the IP is dynamic, that's no big deal.

**Comment by arandomgstring** [comment]
@burntheplanet 
Since when has  VPS's IP become dynamic?

**Comment by wkrp** [comment]
In Iran, there's documentation of SSH being throttled as far back as 2013. The specific observations are likely to be obsolete today, but it shows that circumvention has been more complicated than just using an SSH proxy for many years.

https://censorbib.nymity.ch/#Aryan2013a

> For HTTP and HTTPS file transfers, Aryan used 85% and 89% of its total bandwidth on average, respectively. In contrast, for SSH file transfers, only 15% was utilized on average. All of our measurements were within 5% of these averages.
>
> To confirm that this decrease resulted from the censor’s interference, we proceeded to obfuscate our SSH file transfer (and therefore the unencrypted portion of its handshake) by XORing packet payloads with a predefined constant key. In this way, we expected to circumvent the censor’s efforts to detect and throttle our SSH tunnel. Surprisingly, all of the trials using this modified SSH tunnel exhibited even worse performance. The obfuscated connection was constantly throttled to the point that download speed dropped to near zero at around 60 seconds into the connection. This resulted in incomplete file transfers during all of our trials.
>
> From this observation, we hypothesize that instead of blacklisting undesired protocols, the censorship system was configured to whitelist approved protocols.

**Comment by burntheplanet** [comment]
@arandomgstring I have an Amazon Lightsail instance, and those have a dynamic IP by default if you don't attach any static IP to them. Even if you do, the static IP can be changed just by removing and attaching another one, which can be done for free.

**Comment by burntheplanet** [comment]
@wkrp Yeah, I also heard from a few of my friends that SSH is throttled for them. But for me, I am getting speeds as much as 90 Mb/s, which is even higher than my native internet speed. Just for statistical purposes, it doesn't seem to be limited on Irancell yet.

**Comment by Hadi-1624** [comment]
@burntheplanet  may i ask what client you use on your phone forthis purpose?

**Comment by iminsightman** [comment]
i have been using SSH tunnels as a second layer of security for shadowsocks, over the past few month. it's not perfect but it can outperform V2Ray+TLS easily. Having a second VPS inside Iran, and reverse tunneling to it is still the most flawless configuration i know, all the data is being transported to outside through reverse connections (initiated from outside) (as far as my experience goes, the direction of connections matter to the routers a lot)

The throttling is somewhat present i think, Irancell tampers with the SSH port (22), and i don't know why. there is no significant bandwidth limitation over SSH or at least i haven't encountered any. but, there are lot's of connection drops. i usually have jobs running every few minutes to renew reverse tunnel connections.

**Comment by arandomgstring** [comment]
@iminsightman

Why do you use SSH over shadowsocks, when SSH itself is secure enough?

**Comment by iminsightman** [comment]
@arandomgstring I need a bit of user management functionality, it's not about its security, SSH is not designed to serve as a proxy tool, and sharing it to friends and family is kind of a mess.
Now about Shadowsocks, you could use anything else, maybe VMess+SSH instead of VMess+TLS? DPI has reached some intense levels in the past few month, just a bare protocol can't survive inside the networks, the packets are easily classified as Shadowsocks or VMESS or etc. packets. sometimes the IP address of the server is blocked just after a few days of VPN traffic flowing towards it. and that is exactly why VMESS+TLS has become so popular.
So i'm using SSH, like TLS when used along with VMESS or VLESS, TLS is too much, high latency, long lasting handshakes, and the domain and certificate kind of questions the whole security idea behind it. I was looking at some V2Ray configs my friend bought the other day, i submitted the domain used for TLS to a WHOIS query, the whole identity of the owner popped up, even the phone number!

If its just you using, i think bare SSH as a proxy might deliver a very good job.
